src/assets.rs

```diff
@@ -289,13 +288,17 @@ impl HighlightingAssets {
                 None => {
                     let file_name = path.file_name().unwrap_or_default();
                     self.get_extension_syntax(file_name)?
-                        .or(line_syntax)
                         .ok_or_else(|| Error::UndetectedSyntax(path.to_string_lossy().into()))
                 }
             }
         } else {
             // If a path wasn't provided, we fall back to the detect first-line syntax.
```

Comment:
this comment feels a little out of place now - the fallback is done a few lines further down

Reply:
agree, will fix